x-pack/test_serverless/shared/services/alerting_api.ts

```diff
@@ -0,0 +1,1032 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import type {
+  AggregationsAggregate,
+  SearchResponse,
+} from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import type { Client } from '@elastic/elasticsearch';
+import { MetricThresholdParams } from '@kbn/infra-plugin/common/alerting/metrics';
+import { ThresholdParams } from '@kbn/observability-plugin/common/custom_threshold_rule/types';
+import { v4 as uuidv4 } from 'uuid';
+import type { TryWithRetriesOptions } from '@kbn/ftr-common-functional-services';
+import { RoleCredentials } from '.';
+import type { SloBurnRateRuleParams } from '../../api_integration/services/slo_api';
+import { FtrProviderContext } from '../../functional/ftr_provider_context';
+
+interface CreateEsQueryRuleParams {
+  size: number;
+  thresholdComparator: string;
+  threshold: number[];
+  timeWindowSize?: number;
+  timeWindowUnit?: string;
+  esQuery?: string;
+  timeField?: string;
+  searchConfiguration?: unknown;
+  indexName?: string;
+  excludeHitsFromPreviousRun?: boolean;
+  aggType?: string;
+  aggField?: string;
+  groupBy?: string;
+  termField?: string;
+  termSize?: number;
+  index?: string[];
+}
+const RETRY_COUNT = 10;
+const RETRY_DELAY = 1000;
+
+export function AlertingApiProvider({ getService }: FtrProviderContext) {
+  const retry = getService('retry');
+  const es = getService('es');
+  const requestTimeout = 30 * 1000;
+  const retryTimeout = 120 * 1000;
+  const logger = getService('log');
+  const svlCommonApi = getService('svlCommonApi');
+  const supertestWithoutAuth = getService('supertestWithoutAuth');
```

Comment:
Awesome! It should relatively easy to merge this file together with https://github.com/elastic/kibana/blob/main/x-pack/test/api_integration/deployment_agnostic/services/alerting_api.ts for consistency in serverless and stateful context as followup

Reply:
https://github.com/elastic/kibana/blob/main/x-pack/test/api_integration/deployment_agnostic/services/alerting_api.ts#L68

Reply:
> btw svlCommonApi can be replaced with samlAuth to get internal/common headers in deployment-agnostic context.

Very good point!  Thanks D

Reply:
@dmlemeshko Added follow up ticket to our private board